test/powershell/engine/HelpSystem.OnlineHelp.Tests.ps1

```diff
@@ -0,0 +1,68 @@
+Describe 'Online help tests for PowerShell Core Cmdlets' -Tags "CI" {
+
+    $skipTests = [System.Management.Automation.Platform]::IsIoT -or [System.Management.Automation.Platform]::IsNanoServer -or [System.Management.Automation.Platform]::IsLinux
+
+    BeforeAll {
+
+        if (-not $skipTests)
+        {
+            # Verify that test resouces are available
+            $testFiles = @("$PSScriptRoot\testdata\HelpURI\V2Cmdlets.csv", "$PSScriptRoot\testdata\HelpURI\V3Cmdlets.csv")
+
+            # Enable the test hook
+            [system.management.automation.internal.internaltesthooks]::SetTestHook('BypassOnlineHelpRetrieval', $true)
+
+            # Remove the help content
+            $path = Join-Path $PSHOME (Get-UICulture).Name
+            Get-ChildItem "$path\*dll-help.xml" -Recurse | Remove-Item -Force -ErrorAction SilentlyContinue
+        }
+    }
+
+    AfterAll {
+        
+        if (-not $skipTests)
+        {
+            # Disable the test hook
+            [system.management.automation.internal.internaltesthooks]::SetTestHook('BypassOnlineHelpRetrieval', $false)
+        }
+    }
+
+    $testFiles | foreach {
+        $cmdletList = Import-Csv $_ -ea Stop
+
+        foreach ($cmdlet in $cmdletList)
+        {
+            # If the cmdlet is not preset in CoreCLR, skip it.
+            if (Get-Command $cmdlet.TopicTitle -ea SilentlyContinue)
+            {
+                # TopicTitle - is the cmdlet name in the csv file
+                # HelpURI - is the expected help URI in the csv file
+
+                It "Validate 'get-help $($cmdlet.TopicTitle) -Online'" -skip:($skipTests) {
+                    $actualURI = Get-Help $cmdlet.TopicTitle -Online
+                    $actualURI = $actualURI.Replace("Help URI:","").Trim()
+                    
+                    # helpURI links have been change from http to https, so we need
+                    # to take this into consideration
+                    $actualURI | Should Be $cmdlet.HelpURI.Replace("http", "https").Trim()
+                }
+            }
+        }
+    }
+
+    It "Get-help -online <cmdletName> throws InvalidOperation on IoT and Nano Server" `
+        -skip:(-not ([System.Management.Automation.Platform]::IsIoT -or [System.Management.Automation.Platform]::IsNanoServer)) {
+
+        $myError = $null
+        try
+        {
+            Get-Help Get-Help -Online
+        }
+        catch
+        {
+            $myError = $_
+        }
+
+        $myError.FullyQualifiedErrorId | Should Be "InvalidOperation,Microsoft.PowerShell.Commands.GetHelpCommand"
```

Comment:
since you're now sure that the try will throw, this can become:
```powershell
catch
{
    $_.FullyQualifiedErrorId | Should be "InvalidOperation,Microsoft.PowerShell.Commands.GetHelpCommand"
}
```